src/core/thread/mle.cpp

```diff
@@ -1438,7 +1452,28 @@ void Mle::HandleAttachTimer(Timer &aTimer)
 
 void Mle::HandleAttachTimer(void)
 {
-    uint32_t delay = 0;
+    uint32_t delay          = 0;
+    bool     shouldAnnounce = true;
+
+    if (mAttachState == kAttachStateParentRequestRouter || mAttachState == kAttachStateParentRequestReed ||
+        mAttachState == kAttachStateAnnounce || mAttachState == kAttachStateChildIdRequest)
+    {
```

Comment:
It's possible but practically it will not be likely. When we enter the `kAttachStateAnnounce` we sent a Parent Request to routers/reeds, and then start sending MLE Announcements (basically using the time in this state for two purposes listening for Parent Response and also sending MLE Announcements), so we can possibly get a Parent Response while in this state.